client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -162,25 +165,50 @@ const Checkout = React.createClass( {
 	},
 
 	getCheckoutCompleteRedirectPath: function() {
-		var renewalItem,
+		var product,
+			purchasedProducts,
+			renewalItem,
+			receipt = this.props.transaction.step.data,
 			receiptId = ':receiptId';
 
 		this.props.clearPurchases();
 
 		if ( cartItems.hasRenewalItem( this.props.cart ) ) {
 			renewalItem = cartItems.getRenewalItems( this.props.cart )[ 0 ];
+			// group all purchases into an array
+			purchasedProducts = reduce( receipt && receipt.purchases || {}, function( result, value ) {
+				return result.concat( value );
+			}, [] );
+			// and take the first product which matches the product id of the renewalItem
+			product = find( purchasedProducts, function( item ) {
```

Comment:
Thx, I think I've seen it somewhere 👍  
But for those kind of functions which are also available as part of [Array's prototype](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) in many impl, I like to use the common format (in case we want to get rid of lodash one day).